src/enzo/ActiveParticle_GalaxyParticle.C

```diff
@@ -0,0 +1,445 @@
+//    	      xx = thisGrid->CellLeftEdge[0][i] + 0.5*thisGrid->CellWidth[0][i];
+//           yy = thisGrid->CellLeftEdge[1][j] + 0.5*thisGrid->CellWidth[1][j];
+//           zz = thisGrid->CellLeftEdge[2][k] + 0.5*thisGrid->CellWidth[2][k];
+//           temp = (xx - x) * (xx - x) + (yy - y) * (yy - y) + (zz - z) * (zz - z);
+//           if (temp < dist) {
+//             dist = temp;
+//              mark = index;
+//           }
+//         }
+//       }
+//     }
+// 	tvel = thisGrid->AveragedVelocityAtCell(mark ,data.DensNum,data.Vel1Num);
+	
+	np->vel[0] = vx;
+	np->vel[1] = vy;
+	np->vel[2] = vz;
+	
+	np->Radius = radius;
+	np->Metallicity = 0.0;
+	np->initialized = 0;
+	
+	// Clean up
+	//delete [] tvel;
+  } // end while fgets...
+
+  return SUCCESS;
+}
+
+int ActiveParticleType_GalaxyParticle::EvaluateFeedback
+(grid *thisgrid_orig, ActiveParticleFormationData &data)
+{
+  // Feedback takes place in grid::ApplyGalaxyParticleFeedback.
+
+  return SUCCESS;
+}
+
+void ActiveParticleType_GalaxyParticle::DescribeSupplementalData
+(ActiveParticleFormationDataFlags &flags)
+{
+  flags.DarkMatterDensity = true;
+  flags.Temperature = true;
+  flags.MetalField = true;
+}
+
+int ActiveParticleType_GalaxyParticle::SetFlaggingField
+(LevelHierarchyEntry *LevelArray[], int level,
+							int TopGridDims[], int GalaxyParticleID)
+{
+  /* Generate a list of all galaxy particles in the simulation box */
+  int i, nParticles;
+  FLOAT *pos = NULL, dx=0, rad;
+  ActiveParticleList<ActiveParticleType> GalaxyParticleList;
+  LevelHierarchyEntry *Temp = NULL;
+  
+  ActiveParticleFindAll(LevelArray, &nParticles, GalaxyParticleID, 
+      GalaxyParticleList);
+  
+  /* Calculate CellWidth on maximum refinement level */
+  
+  // this will fail for noncubic boxes or simulations with MinimimMassForRefinementLevelExponent
+  dx = (DomainRightEdge[0] - DomainLeftEdge[0]) /
+    (TopGridDims[0]*POW(FLOAT(RefineBy),FLOAT(MaximumRefinementLevel)));
+  
+  for (i=0 ; i<nParticles; i++){
+    pos = GalaxyParticleList[i]->ReturnPosition();
+    rad = static_cast<ActiveParticleType_GalaxyParticle*>(
+        GalaxyParticleList[i])->Radius;
+    for (Temp = LevelArray[level]; Temp; Temp = Temp->NextGridThisLevel)
+      if (Temp->GridData->DepositRefinementZone(level,pos,rad) == FAIL) {
+	ENZO_FAIL("Error in grid->DepositRefinementZone.\n")
+	  }
+  }
+
+  return SUCCESS;
+}
+
+grid* ConstructFeedbackZone(ActiveParticleType* ThisParticle, int FeedbackRadius, 
+			    FLOAT dx, HierarchyEntry** Grids, int NumberOfGrids,
+			    int SendField);
+
+int DistributeFeedbackZone(grid* FeedbackZones, HierarchyEntry** Grids, 
+			   int NumberOfGrids, int SendField);
+
+int ActiveParticleType_GalaxyParticle::SubtractMassFromGrid(int nParticles,
+    ActiveParticleList<ActiveParticleType>& ParticleList, LevelHierarchyEntry *LevelArray[],
+    FLOAT dx, int ThisLevel)
+{
+
+  // This function is a work in progress.
+  return SUCCESS;
+
```

Comment:
This function has all the arguments commented out and is not being used. It should be removed.